javascript/javascript3/week2/promise-visual/main.js

```diff
@@ -0,0 +1,35 @@
+//Visual Promise
+//One By One
+const span = document.querySelector('span');
+async function translateOneByOne(nthChild, x, y){
+    return await new Promise (resolve => {
+        moveElement(document.querySelector(`li:nth-child(${nthChild})`), { 
+            x, 
+            y 
+        })
+        .then(() => {
+            console.log("Element has been moved");
+            resolve();
+        });
+    })
+}
+
+translateOneByOne(1, 20, 300)
+.then(() => translateOneByOne(2, 400, 300))
+.then(() => translateOneByOne(3, 400, 20))
+.then (() => span.setAttribute('class', 'shown'))
```

Comment:
Good job.
Some other ways just to take a lock
```js

const [red, blue, green] = document.querySelectorAll('.marks li');
const target = {
  red: { x: 20, y: 300 },
  blue: { x: 400, y: 300 },
  green: { x: 400, y: 20 }
};

function translateOneByOne() {
  moveElement(red, target.red)
    .then(() => {
      console.log('red moved');
      return moveElement(blue, target.blue);
    })
    .then(() => {
      console.log('blue moved');
      return moveElement(green, target.green);
    })
    .then(() => {
      console.log('green moved');
    });
}
// translateOneByOne();

async function asyncTranslateOneByOne() {
  await moveElement(red, target.red);
  console.log('red moved');

  await moveElement(blue, target.blue);
  console.log('blue moved');

  await moveElement(green, target.green);
  console.log('green moved');
}

// asyncTranslateOneByOne();
```

Reply:
Thanks for all your comments 🙏👍